cypress/e2e/room/room-header.spec.ts

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2023 Suguru Hirahara
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { HomeserverInstance } from "../../plugins/utils/homeserver";
+import { SettingLevel } from "../../../src/settings/SettingLevel";
+
+describe("Room Header", () => {
+    let homeserver: HomeserverInstance;
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, "Sakura");
+        });
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should render seven buttons by default", () => {
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findAllByRole("button").should("have.length", 7).should("be.visible");
+
+            cy.findByRole("button", { name: "Room options" }).should("be.visible");
+            cy.findByRole("button", { name: "Voice call" }).should("be.visible");
+            cy.findByRole("button", { name: "Video call" }).should("be.visible");
+            cy.findByRole("button", { name: "Search" }).should("be.visible");
+            cy.findByRole("button", { name: "Threads" }).should("be.visible");
+            cy.findByRole("button", { name: "Notifications" }).should("be.visible");
+            cy.findByRole("button", { name: "Room info" }).should("be.visible");
+        });
+
+        cy.get(".mx_RoomHeader").percySnapshotElement("Room header", {
+            widths: [300, 600], // Magic numbers
```

Comment:
Values specified with `widths` are used to specify width of the snapshot of the chained element. The default values are 1024 and 1920 (both in pixel) specified [here](https://github.com/matrix-org/matrix-react-sdk/blob/4c72cceb70700ea9c9a36a938497520ba289dfdd/.percy.yml#L4-L5) on `.percy.yml`.

The custom values are indeed relevant only for `Room header - with a long room name`, since the object of the snapshot with overriding the default values should be to emulate the narrow UI and ensure that room name overflow is properly hidden with horizontal ellipsis. I'm going to remove unrelated widths settings.


I'm going to remove unrelated widths settings.

Reply:
Addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10713/commits/2922c66fe27ceff6fd253d4a1d842f808dd2a37a.